README.md

```diff
@@ -85,6 +88,165 @@ public class TokenGenerationUtil {
+            System.out.println(bearerToken);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        // Generate BearerToken with context by specifying credentials.json as string
+        try {
+            String fileContents = "<YOUR_CREDENTIALS_FILE_CONTENTS_AS_STRING>";
+            BearerToken token = new BearerToken.BearerTokenBuilder()
+                    .setCredentials(fileContents)
+                    .setContext("abc")
+                    .build();
+
+            bearerToken = token.getBearerToken();
+            System.out.println(bearerToken);
+
+        } catch (SkyflowException e) {
+            e.printStackTrace();
+        }
+
+    }
+}
+
+```
+
+Note: 
+- Either credentials file path or credentials as string can be passed to `setCredentials` method of BearerTokenBuilder class.
+- If both credentials file path and credentials string are passed to `setCredentials` method, then setter which is at the last will be taken precedence over the other.
+
+## Service Account Scoped Bearer Token Generation
+
+A service account having multiple roles can be used to generate bearer tokens with restricted access by providing the appropriate `roleID`. The token generated is valid for 60 minutes and perform opertaions with permissions associated with the role.
+
+[Example](https://github.com/skyflowapi/skyflow-java/blob/master/samples/src/main/java/com/example/ScopedTokenGeneration.java):
+
+```java
+import java.io.File;
+
+public class ScopedTokenGeneration {
+    public static void main(String args[]) {
+
+        String scopedToken = null;
+
+        // Generate Scoped Token  by specifying credentials.json file path
+        try {
+            String filePath = "<YOUR_CREDENTIALS_FILE_PATH>";
+            BearerToken token = new BearerToken.BearerTokenBuilder()
+                    .setCredentials(new File(filePath))
+                    .setRoles(new String[]{"roleID"})
+                    .build();
+
+            scopedToken = token.getBearerToken();
+            System.out.println(scopedToken);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+}
+```
+Note: 
+- Either credentials file path or credentials as string can be passed to `setCredentials` method of BearerTokenBuilder class.
+- If both credentials file path and credentials string are passed to `setCredentials` method, then setter which is at the last will be taken precedence over the other.
+
+## Signed Data Tokens Generation
+
+Skyflow data tokens are generated when sensitive data is inserted into the vault. These data tokens can be digitally signed with the private key of the service account credentials provided, which adds an additional layer of protection. These signed tokens can then be detokenized by passing a signed data token and a bearer token generated from service account credentials having appropriate permissions and context to detokenize API calls.
```

Comment:
These data tokens can be digitally signed with the private key of the service account credentials, which adds an additional layer of protection. Signed tokens can be detokenized by passing the signed data token and a bearer token generated from service account credentials. The service account must have appropriate permissions and context to detokenize the signed data tokens.